src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+        const importSummaries = [testImportSummary(), testImportSummary()]
+        const anExchange = testDataExchange({
+            requests: [testRequest(), testRequest()],
+        })
+
+        const exchanges = [anExchange]
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+            importSummaryResponse: { importSummaries },
+        })
+
+        const bottomBar = await screen.findByTestId('bottom-bar')
+
+        const submitButton = within(bottomBar).getByRole('button', {
+            name: 'Submit data',
+        })
+        submitButton.click()
+
+        const submitModal = await screen.findByTestId('dhis2-uicore-card')
+        within(submitModal).getByTestId('confirm-submission-button').click()
+
+        waitFor(() => expect(submitButton).toBeDisabled())
+        const successTag = await screen.findByTestId('dhis2-uicore-tag')
+        expect(successTag).toBeInTheDocument()
+
+        const summaryBoxes = screen.getAllByTestId('summary-box')
+        expect(summaryBoxes).toHaveLength(3)
+        expect(summaryBoxes[0]).toHaveTextContent(
+            `${
+                importSummaries[0].importCount.imported +
+                importSummaries[1].importCount.imported
+            }imported`
+        )
+        expect(summaryBoxes[1]).toHaveTextContent(
+            `${
+                importSummaries[0].importCount.updated +
+                importSummaries[1].importCount.updated
+            }updated`
+        )
+        expect(summaryBoxes[2]).toHaveTextContent(
+            `${
+                importSummaries[0].importCount.ignored +
+                importSummaries[1].importCount.ignored
+            }ignored`
+        )
+
+        const summariesRows = within(submitModal).getAllByTestId(
+            'dhis2-uicore-datatablerow'
+        )
+        expect(summariesRows).toHaveLength(3)
+        expect(summariesRows[1]).toHaveTextContent(
+            anExchange.source.requests[0].name
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.imported
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.updated
+        )
+        expect(summariesRows[1]).toHaveTextContent(
+            importSummaries[0].importCount.ignored
+        )
+
+        expect(summariesRows[2]).toHaveTextContent(
+            anExchange.source.requests[1].name
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.imported
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.updated
+        )
+        expect(summariesRows[2]).toHaveTextContent(
+            importSummaries[1].importCount.ignored
+        )
+    })
+
+    it('show an error when a exchange data is submitted un-successfully', async () => {
```

Comment:
I guess it would be also nice to have a test that covers the scenario where the submit itself fails (I suppose by modifying the mock return of useDataEngine?). 

The post to api/aggregateDataExchanges/${id}/exchange  is a bit weird in the sense that it can come back as 200 but tell you that there were errors trying when the server submitted, or there can be a 400-range error if the server fails to make the submit.

There's also a bunch of test scenarios in the cypress tests that cover some of the behavior for the submit button  that might be better as unit tests (like when you're allowed to resubmit, etc.,)